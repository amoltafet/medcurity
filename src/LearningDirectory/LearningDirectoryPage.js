import { Row} from 'react-bootstrap'
import React from 'react';
import 'bootstrap/dist/css/bootstrap.min.css'
import { useEffect, useState } from "react";
import { useParams } from "react-router";
import axios from 'axios';
import LearningModulePanel from './LearningModulePanel';
import './LearningDirectory.css'
// import env from "react-dotenv";

/**
 * Constructs the content for the Learning Directory Page
 * @param {DirectoryTitle}: props
 * @param {modules}: props
 * @returns Page content of the directory page
 */
const LearningDirectoryPageContent = (props) => {

    /**
     * Create directory cards from modules
     */
    function createDirectoryCards(modules) {
        var dueDate = new Date(props.dueDate); 
        const objs = []
        for (let index in modules) {
            var newModule = modules[index]
            objs.push(<LearningModulePanel title={newModule.Title} link={newModule.ID} dueDate={newModule.DueDate}/>)
        }
        return objs;
    }

    return (
        <>
         
        <Row className="dashboard" style={{display: 'flex', flexDirection: 'row'}}>
            {createDirectoryCards(props.modules)}
        </Row>
        <div className="d-grid gap-2">
        </div>
        </>
    );
}

/**
* Creates and displays the learning directory for each category. 
* @return { LearningDirectoryPage}
*/
const  LearningDirectoryPage = () => {
    let { slug } = useParams();
    const [directory, setDirectory] = useState([])
    const [modules, setModules] = useState([])

    useEffect(() => {
      // Fetch post using the postSlug
    }, [slug]);

    // Query for getting LearningModules Directory info
    useEffect(() => {
        axios.get(`${process.env.REACT_APP_BASE_URL}/api/getQuery`, { params: { the_query: "SELECT * FROM LearningModulesDirectory WHERE ID = " + slug } }).then((response) => {
              setDirectory(Object.values(response.data))
        }).catch(error => console.error(`Error ${error}`));
    }, [slug])

    // Query for getting info on learning modules associated with the directory
    useEffect(() => {
        axios.get(`${process.env.REACT_APP_BASE_URL}/api/getQuery`, { params: { the_query:"SELECT * FROM LearningModules WHERE DirID = " + slug} }).then((response) => {
            setModules(Object.values(response.data))
        }).catch(error => console.error(`Error ${error}`));
    }, [slug])

    const directoryTitle = directory.map((directory) => {
        return directory.Title
    })

    return(
        <LearningDirectoryPageContent directoryTitle={directoryTitle +  ' Learning Modules Directory'} modules={modules} />
    );


}

export { LearningDirectoryPageContent }
export default  LearningDirectoryPage;