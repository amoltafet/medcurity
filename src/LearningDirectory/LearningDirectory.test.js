import renderer from 'react-test-renderer';
import LearningDirectoryPage from './LearningDirectory';
import React from 'react';
import { BrowserRouter } from 'react-router-dom';

test('renders LearningDirectoryPage', () => {
  const tree = renderer.create(<BrowserRouter><LearningDirectoryPage /></BrowserRouter>).toJSON();
  expect(tree).toMatchSnapshot();
});