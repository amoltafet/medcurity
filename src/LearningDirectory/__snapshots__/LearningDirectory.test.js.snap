// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders LearningDirectoryPage 1`] = `
Array [
  <nav
    class="navbar navbar-expand-lg navbar-dark bg-dark"
  >
    <div
      class="container-fluid"
    >
      <a
        href="/dash"
      >
        <img
          alt=""
          className="card-img-top MedcurityLogo mr-5"
          src="/Medcurity_Logo.png"
        />
      </a>
      <button
        aria-controls="navbarSupportedContent"
        aria-expanded="false"
        aria-label="Toggle navigation"
        class="navbar-toggler"
        data-mdb-target="#navbarSupportedContent"
        data-mdb-toggle="collapse"
        type="button"
      >
        <i
          class="fas fa-bars text-light"
        />
      </button>
      <div
        class="collapse navbar-collapse"
        id="navbarSupportedContent"
      >
        <ul
          class="navbar-nav me-auto d-flex flex-row mt-3 mt-lg-0 "
        >
          <li
            class="nav-item text-center mx-2 mx-lg-1"
          >
            <a
              aria-current="page"
              class="nav-link active"
              href="#!"
            >
              <div>
                <svg
                  class="bi bi-house-door"
                  fill="currentColor"
                  height="16"
                  viewBox="0 0 16 16"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M8.354 1.146a.5.5 0 0 0-.708 0l-6 6A.5.5 0 0 0 1.5 7.5v7a.5.5 0 0 0 .5.5h4.5a.5.5 0 0 0 .5-.5v-4h2v4a.5.5 0 0 0 .5.5H14a.5.5 0 0 0 .5-.5v-7a.5.5 0 0 0-.146-.354L13 5.793V2.5a.5.5 0 0 0-.5-.5h-1a.5.5 0 0 0-.5.5v1.293L8.354 1.146zM2.5 14V7.707l5.5-5.5 5.5 5.5V14H10v-4a.5.5 0 0 0-.5-.5h-3a.5.5 0 0 0-.5.5v4H2.5z"
                  />
                </svg>
              </div>
              Home
            </a>
          </li>
          <li
            class="nav-item text-center mx-2 mx-lg-1"
          >
            <a
              class="nav-link"
              href="/dash#leaderboard"
            >
              <div>
                <svg
                  class="bi bi-clipboard2-data"
                  fill="currentColor"
                  height="16"
                  viewBox="0 0 16 16"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M9.5 0a.5.5 0 0 1 .5.5.5.5 0 0 0 .5.5.5.5 0 0 1 .5.5V2a.5.5 0 0 1-.5.5h-5A.5.5 0 0 1 5 2v-.5a.5.5 0 0 1 .5-.5.5.5 0 0 0 .5-.5.5.5 0 0 1 .5-.5h3Z"
                  />
                  <path
                    d="M3 2.5a.5.5 0 0 1 .5-.5H4a.5.5 0 0 0 0-1h-.5A1.5 1.5 0 0 0 2 2.5v12A1.5 1.5 0 0 0 3.5 16h9a1.5 1.5 0 0 0 1.5-1.5v-12A1.5 1.5 0 0 0 12.5 1H12a.5.5 0 0 0 0 1h.5a.5.5 0 0 1 .5.5v12a.5.5 0 0 1-.5.5h-9a.5.5 0 0 1-.5-.5v-12Z"
                  />
                  <path
                    d="M10 7a1 1 0 1 1 2 0v5a1 1 0 1 1-2 0V7Zm-6 4a1 1 0 1 1 2 0v1a1 1 0 1 1-2 0v-1Zm4-3a1 1 0 0 0-1 1v3a1 1 0 1 0 2 0V9a1 1 0 0 0-1-1Z"
                  />
                </svg>
              </div>
              Leaderboard
            </a>
          </li>
          <li
            class="nav-item text-center mx-2 mx-lg-1"
          >
            <a
              aria-current="page"
              class="nav-link"
              href=""
            >
              <div>
                <svg
                  class="bi bi-journal-bookmark"
                  fill="currentColor"
                  height="16"
                  viewBox="0 0 16 16"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M6 8V1h1v6.117L8.743 6.07a.5.5 0 0 1 .514 0L11 7.117V1h1v7a.5.5 0 0 1-.757.429L9 7.083 6.757 8.43A.5.5 0 0 1 6 8z"
                    fill-rule="evenodd"
                  />
                  <path
                    d="M3 0h10a2 2 0 0 1 2 2v12a2 2 0 0 1-2 2H3a2 2 0 0 1-2-2v-1h1v1a1 1 0 0 0 1 1h10a1 1 0 0 0 1-1V2a1 1 0 0 0-1-1H3a1 1 0 0 0-1 1v1H1V2a2 2 0 0 1 2-2z"
                  />
                  <path
                    d="M1 5v-.5a.5.5 0 0 1 1 0V5h.5a.5.5 0 0 1 0 1h-2a.5.5 0 0 1 0-1H1zm0 3v-.5a.5.5 0 0 1 1 0V8h.5a.5.5 0 0 1 0 1h-2a.5.5 0 0 1 0-1H1zm0 3v-.5a.5.5 0 0 1 1 0v.5h.5a.5.5 0 0 1 0 1h-2a.5.5 0 0 1 0-1H1z"
                  />
                </svg>
              </div>
              Modules
            </a>
          </li>
        </ul>
        <ul
          class="navbar-nav ms-auto d-flex flex-row mt-3 mt-lg-0"
        >
          <div
            class="vertical-line"
          />
          <li
            class="nav-item text-center mx-2 mx-lg-1"
          >
            <a
              class="nav-link"
              href="/notifications"
            >
              <div>
                <svg
                  class="bi bi-bell"
                  fill="currentColor"
                  height="24"
                  viewBox="0 0 16 16"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M8 16a2 2 0 0 0 2-2H6a2 2 0 0 0 2 2zM8 1.918l-.797.161A4.002 4.002 0 0 0 4 6c0 .628-.134 2.197-.459 3.742-.16.767-.376 1.566-.663 2.258h10.244c-.287-.692-.502-1.49-.663-2.258C12.134 8.197 12 6.628 12 6a4.002 4.002 0 0 0-3.203-3.92L8 1.917zM14.22 12c.223.447.481.801.78 1H1c.299-.199.557-.553.78-1C2.68 10.2 3 6.88 3 6c0-2.42 1.72-4.44 4.005-4.901a1 1 0 1 1 1.99 0A5.002 5.002 0 0 1 13 6c0 .88.32 4.2 1.22 6z"
                  />
                </svg>
              </div>
            </a>
          </li>
          <div
            class="vertical-line"
          />
          <div
            class="vertical-line"
          />
          <li
            class="nav-item text-center mx-2 mx-lg-1"
          >
            <button
              aria-controls="navbar-dark-example"
              aria-label="Toggle navigation"
              class="toggle"
              className="navbar-toggler collapsed"
              onClick={[Function]}
              type="button"
            >
              <span
                className="navbar-toggler-icon"
              />
            </button>
            <div
              aria-expanded={null}
              className="navbar-collapse collapse"
              id="navbar-dark-example"
            >
              <div
                className="nav"
                onKeyDown={[Function]}
              >
                <div
                  bsStyle="dark"
                  className="nav-item dropdown"
                  menuVariant="dark"
                  noCaret={true}
                  size="lg"
                >
                  <a
                    aria-expanded={false}
                    aria-haspopup={true}
                    className="dropdown-toggle nav-link"
                    data-rb-event-key={null}
                    href="#"
                    id="nav-dropdown-dark-example"
                    onClick={[Function]}
                    onKeyDown={[Function]}
                    role="button"
                  >
                    <svg
                      class="bi bi-person-square"
                      fill="currentColor"
                      height="24"
                      viewBox="0 0 16 16"
                      width="24"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M11 6a3 3 0 1 1-6 0 3 3 0 0 1 6 0z"
                      />
                      <path
                        d="M2 0a2 2 0 0 0-2 2v12a2 2 0 0 0 2 2h12a2 2 0 0 0 2-2V2a2 2 0 0 0-2-2H2zm12 1a1 1 0 0 1 1 1v12a1 1 0 0 1-1 1v-1c0-1-1-4-6-4s-6 3-6 4v1a1 1 0 0 1-1-1V2a1 1 0 0 1 1-1h12z"
                      />
                    </svg>
                  </a>
                </div>
              </div>
            </div>
          </li>
          <div
            class="vertical-line"
          />
          <div>
            <form
              class="d-flex input-group w-auto ms-lg-3 my-3 my-lg-0"
            >
              <input
                aria-label="Search"
                class="form-control"
                onChange={[Function]}
                placeholder="Search Modules"
                type="search"
              />
              <button
                class="btn search-btn-color"
                data-mdb-ripple-color="dark"
                type="button"
              >
                <svg
                  class="bi bi-search"
                  fill="currentColor"
                  height="16"
                  viewBox="0 0 16 16"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M11.742 10.344a6.5 6.5 0 1 0-1.397 1.398h-.001c.03.04.062.078.098.115l3.85 3.85a1 1 0 0 0 1.415-1.414l-3.85-3.85a1.007 1.007 0 0 0-.115-.1zM12 6.5a5.5 5.5 0 1 1-11 0 5.5 5.5 0 0 1 11 0z"
                  />
                </svg>
              </button>
            </form>
          </div>
        </ul>
      </div>
    </div>
  </nav>,
  <div
    className="home"
  >
    <h1
      className="text-center mt-5"
    >
       
    </h1>
  </div>,
]
`;
